06_puppy/nicholascross/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+type ID uint64
+type Breed string
+type Colour string
+type Value float64
+
+type Puppy struct {
+	ID     ID
+	Breed  Breed
+	Colour Colour
+	Value  Value
+}
+
+type Storer interface {
+	CreatePuppy(p Puppy)
+	RetrievePuppy(ID) *Puppy
+	UpdatePuppy(ID, Puppy)
+	DeletePuppy(ID)
+}
+
+type MapStore map[ID]*Puppy
```

Comment:
I'd advise against pointers for in memory storage. The reason is that it is safer to handle ([playspace](https://goplay.space/#O4dHwcB5_Ak))